File: JavaScript/88.js
/**
 * @no 88
 * @name Merge Sorted Array
 */
let merge = (nums1, m, nums2, n) => {
    for (let k = 0; k < n; k++) {
        let find = false;
        for (let i = 0; i < m + k; i++) {
            if (nums2[k] <= nums1[i]) {
                for (let j = m + k; j > i; j--)
                    nums1[j] = nums1[j - 1];
                nums1[i] = nums2[k];
                find = true;
                break;
            }
        }
        if (!find) {
            nums1[m + k] = nums2[k];
        }
    }
};

File: Python/88.py
"""
@no 88
@name Merge Sorted Array
"""
class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        for i in range(0, n):
            if m + i == 0:
                nums1[0] = nums2[i]
            if nums2[i] > nums1[m + i - 1]:
                nums1[m + i] = nums2[i]
            else:
                for j in range(0, m + i):
                    if (nums2[i] <= nums1[j]):
                        for k in range(m + i, j, -1):
                            nums1[k] = nums1[k - 1]
                        nums1[j] = nums2[i]
                        break
